docs/site/exposing-graphql-apis.md

```diff
@@ -0,0 +1,112 @@
+---
+lang: en
+title: 'Exposing GraphQL APIs'
+keywords: LoopBack 4.0, LoopBack 4, GraphQL
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/exposing-graphql-apis.html
+---
+
+## Overview
+
+The [OASGraph module](https://www.npmjs.com/package/oasgraph) creates a GraphQL
+wrapper for existing REST APIs which are described by the OpenAPI specification.
+This tutorial shows how to expose GraphQL APIs in an existing LoopBack
+application.
+
+### Prerequisite
+
+Make sure you have a running LoopBack 4 application. In this tutorial, we'll use
+the `todo` example. You can create this application by running the command
+below:
+
+```sh
+lb4 example todo
+```
+
+### Install OASGraph and Required Dependencies
+
+From your LoopBack application, run the following command to install OASGraph
+and the required dependencies:
+
+```sh
+npm i --save oasgraph express express-graphql
+```
+
+### Start the GraphQL Server
+
+Make sure your LoopBack application is running by going to
+`http://localhost:3000/openapi.json`. If not, you can start it by running the
+`npm start` command.
+
+Now we will use the oasgraph CLI to set up a GraphQL HTTP Server backed by
+express on port 3001. Specifying the OpenAPI spec generated by the
+todo-application as the parameter, start up the server by running the following
+command:
+
+```sh
+npx oasgraph http://localhost:3000/openapi.json
+```
+
+If you're using npm version lower than 5.2, run the command below instead:
+
+```sh
+node_modules/.bin/oasgraph http://localhost:3000/openapi.json
+```
+
+That’s it! You’re now ready to try out some tests and requests in the browser at
+http://localhost:3001/graphql.
```

Comment:
Let's start a new conversation focused on the scope of this pull request.

I am proposing to add a short explanation of why GraphQL is served at a different port and point people to #1905 to get updates on our progress towards a better solution.

For example:

```markdown
That’s it! You’re now ready to try out some tests and requests in the browser at
http://localhost:3001/graphql.

{% include note.html content="
We are looking into ways how to expose the GraphQL endpoint alongside the main REST API,
on the same HTTP host and port. See
[issue #1905](https://github.com/strongloop/loopback-next/issues/1905).
" %}
```